examples/docs_snippets/docs_snippets/guides/components/shell-script-component/with-build-defs.py

```diff
@@ -9,17 +9,12 @@
 import dagster as dg
 
 
-@dataclass
+@dataclass(frozen=True)
```

Comment:
although we use `frozen` internally our scaffolding does not product it so it might just be confusing.

Reply:
Rebased your changes, and removed the `frozen=True`.

@OwenKephart I also included a note that this is opt-in, with the ability to pass `--no-dataclass`.